eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageBuilder.java

```diff
@@ -0,0 +1,67 @@
+package io.eventuate.messaging.kafka.common;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.Optional;
+
+public class EventuateKafkaMultiMessageBuilder {
+  public static final String MAGIC_ID = "a8c79db675e14c4cbf1eb77d0d6d0f00"; // generated UUID
+  public static final byte[] MAGIC_ID_BYTES = MAGIC_ID.getBytes();
+
+  private int maxSize;
+  private int size;
+  private ByteArrayOutputStream binaryStream = new ByteArrayOutputStream();
+
+  public EventuateKafkaMultiMessageBuilder(int maxSize) {
+    this.maxSize = maxSize;
+
+    try {
+      binaryStream.write(MAGIC_ID_BYTES);
+      size += MAGIC_ID_BYTES.length;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public boolean addMessage(EventuateKafkaMultiMessageKeyValue message) {
+    try {
+      byte[] keyBytes = Optional.ofNullable(message.getKey()).map(String::getBytes).orElse(new byte[0]);
```

Comment:
getBytes() should be passed a CharSet to ensure cross-platform consistency

Reply:
yes, I also thought about, so I will fix it tomorrow everywhere.

Reply:
Similarly, new String(..) or equivalent needs to use the same Charset

Reply:
ok